contracts/0.4.24/lib/StakeLimitUtils.sol

```diff
@@ -145,13 +145,19 @@ library StakeLimitUtils {
             "TOO_SMALL_LIMIT_INCREASE"
         );
 
-        // if staking was paused or unlimited previously,
-        // or new limit is lower than previous, then
-        // reset prev stake limit to the new max stake limit
-        if ((_data.maxStakeLimit == 0) || (_maxStakeLimit < _data.prevStakeLimit)) {
+        // reset prev stake limit to the new max stake limit if
+        if (
+            // staking was paused or
+            _data.prevStakeBlockNumber == 0 ||
+            // staking was unlimited or
+            _data.maxStakeLimit == 0 ||
+            // new limit is lower than the previous
```

Comment:
Comment is misleading here. It make you think that prevStakeLimit is prevMaxStakeLimit, but it's a current level in the Diablo mana sphere

Reply:
yeah, good catch!
ty, fixed :+1: